File: insertion-sort/insertion-sort.js
/**
 * @param {Array} a
 */
var insertionSort = function(a) {
  var i, j;
  for (i = 0; i < a.length; i++) {
    var currentValue = a[i];
    j = i - 1;
    while (j >= 0 && currentValue < a[j]) {
      a[j+1] = a[j];
      j--;
    }
    a[j+1] = currentValue;
  }
};

// Test the algorithm
var input = [1, 0, -1, 10, 40, 4, 2, 3];
insertionSort(input);
console.log(input);

File: insertion-sort/insertion-sort.py
'''
Example of an implementation of insertion sort in Python
'''

def insertionSort(a):
    for i in range(len(a)):
        currentValue = a[i]
        j = i - 1
        while j >= 0 and a[j] > currentValue:
            a[j+1] = a[j]
            j = j - 1
        a[j+1] = currentValue

# Test the algorithm
input = [1, 0, -1, 10, 40, 4, 2, 3];
insertionSort(input);
print input
